test_parser.py

```diff
@@ -0,0 +1,85 @@
+from lexer import Lexer
+from dagql_parser import Parser
+from tokens import TokenTypes, Token
+
+
+def test_basic_select():
+    input_program = """
+    SELECT * FROM EDGES
+    """
+
+    ast = Parser(Lexer(input_program).lex()).parse()
+    assert(ast[0].subquery == Token(TokenTypes.KW, "EDGES", 2))
+
+def test_muti_statement_program():
+    input_program = """
+    SELECT * FROM EDGES
+    SELECT * FROM NODES
+    """
+
+    ast = Parser(Lexer(input_program).lex()).parse()
+    assert(ast[0].subquery == Token(TokenTypes.KW, "EDGES", 2))
+    assert(ast[1].subquery == Token(TokenTypes.KW, "NODES", 3))
+
+def test_nested_select():
+    input_program = """
+    SELECT a FROM (SELECT b FROM NODES)
+    """
+
+    ast = Parser(Lexer(input_program).lex()).parse()
+    assert(ast[0].args[0].token == Token(TokenTypes.ID, 'a', 2))
+
+def test_WHERE_WITH_TRUE():
+    input_program = """
+    SELECT a FROM EDGES WHERE TRUE
+    """
+    ast = Parser(Lexer(input_program).lex()).parse()
```

Comment:
If I'm not mistaken, all of the tests past this point don't assert anything?

Reply:
Yeah it doesn't right now. I had some trouble traversing the ast since I don't know what structure to expect yet. I just want to make sure I cover all the test cases first and slowly adding the assert to each test cases.